cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/query/IndexCreator.java

```diff
@@ -355,4 +371,47 @@ private String createVirtualTableStatementForIndex(String indexName,
                                                                               settings);
     }
 
+    /**
+     * Iterate candidate indexNames generated from the indexNameRandom generator
+     * until we find one which doesn't already exist.
+     *
+     * We make sure the generated name is not an index already by the list
+     * of index names returned by {@link #listIndexesInDatabaseQueue()} method.
+     * This is because we avoid knowing about how indexes are stored in SQLite, however
+     * this means that it is not thread safe, it is possible for a new index with the same
+     * name to be created after a copy of the indexes has been taken from the database.
+     *
+     * We allow up to 200 random name generations, which should give us many millions
+     * of indexes before a name fails to be generated and makes sure this method doesn't
+     * loop forever.
+     *
+     * @return The generated index name or {@code null} if it failed.
+     *
+     */
+    private String generateIndexName() throws ExecutionException, InterruptedException {
```

Comment:
Can this be `private static string generateIndexName(Set existingNames)`? For me, it'd be nice if this method didn't do any database access -- make it a pure function with fewer side-effects.